### PR TITLE
Add cases with executable counterexamples

### PR DESCRIPTION
**Pull request by danblitzhou**
Created test cases that return `sat` under verification and produce executable counterexamples.
This is done by:
1. modifying existing test harnesses to manually introduce verification bugs;
2. modifying and introduce verification bugs into assertion helpers/models

WIP: 
- [x] create more automation in `verify` script or cmake to streamline cex generation.

### CODE REVIEW

**Comment by agurfinkel** [comment]
create directory `cex` along `jobs` and move counterexamples there. They are for us only. When we show the repo to others, it should be clear where the things that they might care about are

File: seahorn/sea_cex_base.yaml [hunk 1]
```diff
@@ -43,8 +43,6 @@ verify_options:
   temp-dir: /tmp/verify-c-common
 # time and result stats
   horn-stats: true
-# produce counterexample harness
-  cex: /tmp/h.ll
```
Comment (agurfinkel) [inline seahorn/sea_cex_base.yaml:47]: why is this removed? cex will not be produced without this option

File: verify.py.in [hunk 2]
```diff
@@ -30,13 +31,31 @@ def check_vacuity(line, passed_set, failed_set):
...
+    if not args.dry_run:
+        subprocess.run(cmd)
 
 def main(argv):
     import sea
 
     def check_vacuity_inner(line, passed_set, failed_set):
         return check_vacuity(line, passed_set, failed_set)
 
+    def generate_cex_exe_inner(argp, input_file, output_file):
```
Comment (agurfinkel) [inline verify.py.in:56]: why `inner`??? 

same question for `check_vacuity_inner`

File: verify.py.in [hunk 3]
```diff
@@ -116,9 +142,11 @@ def main(argv):
                 print(' '.join(cmd))
    
             if args.expect is None:
-                os.execv(cmd[0], cmd)
+                output = subprocess.run(cmd)
```
Comment (agurfinkel) [inline verify.py.in:145]: you just killed my main functionality :(

File: seahorn/sea_cex_base.yaml (hunk 1 above)
Comment (danblitzhou) [reply seahorn/sea_cex_base.yaml:47]: in verify script, I changed the behaviour of cex mode to create harness files with jobs specific naming; if every job outputs cex harness to the same file, during executable generation we might use the wrong harness file

File: verify.py.in [hunk 4]
```diff
@@ -30,13 +31,31 @@ def check_vacuity(line, passed_set, failed_set):
         and m.group('result') == 'failed'):
         failed_set.add(debugInfo.strip())
 
+def generate_cex_exe(args, input_file, output_file):
+    cmd = [os.path.join(SEAHORN_ROOT, 'bin', 'sea'),
```
Comment (agurfinkel) [inline verify.py.in:35]: I don't see how this is workable. Once the harness is produced, it has to be linked with the original input. The users (us) will want to regenerate original input by changing it and running with the same harness. 

The current solution requires re-running build and verification every time the harness is changed.

Let's come up with a better solution and then figure out how to implement it.

**Comment by agurfinkel** [comment]
Maybe we need more time to discuss this.

I can't accept the current PR because:
1. there is too much duplication. It will be very hard to manage, and the overall utility of having all these files is unclear.
2. the changes to verify script are not compatible with our current flow
3. we need to come up and support a workflow for counterexamples. Current solution is geared to running something ones. Not to help a user who is trying to understand what is wrong with either code under analysis, library, or harness.
